tests/test-write-diagnostics.c

```diff
@@ -0,0 +1,287 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+#include "config.h"
+#include <string.h>
+#include <locale.h>
+
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "gis-write-diagnostics.h"
+#include "glnx-fdio.h"
+#include "glnx-shutil.h"
+
+typedef struct {
+  gchar *tmpdir;
+  gchar *image_dir;
+  gchar *home_dir;
+
+  gchar *output;
+  GError *error;
+} Fixture;
+
+static void
+fixture_set_up (Fixture *fixture,
+                gconstpointer user_data)
+{
+  GError *error = NULL;
+
+  fixture->tmpdir = g_dir_make_tmp ("eos-installer.XXXXXX", &error);
+  g_assert_no_error (error);
+  g_assert (fixture->tmpdir != NULL);
+}
+
+static void
+fixture_tear_down (Fixture *fixture,
+                   gconstpointer user_data)
+{
+  g_autoptr(GError) error = NULL;
+
+  if (!glnx_shutil_rm_rf_at (AT_FDCWD, fixture->tmpdir, NULL, &error))
+    g_warning ("Failed to remove %s: %s", fixture->tmpdir, error->message);
+
+  g_clear_pointer (&fixture->tmpdir, g_free);
+  g_clear_pointer (&fixture->image_dir, g_free);
+  g_clear_pointer (&fixture->home_dir, g_free);
+
+  g_clear_pointer (&fixture->output, g_free);
+  g_clear_error (&fixture->error);
+}
+
+static gboolean
+mkdir_ (const char *path,
+        GError    **error)
+{
+  mode_t mode = 0755;
+
+  if (g_mkdir (path, mode) != 0)
+    return glnx_throw_errno_prefix (error, "mkdir(%s, %o)", path, mode);
```

Comment:
Why not use `glnx_ensure_dir (-1, path, 0755, error)`? Does this need to work on WIndows?

Reply:
I did not know that function existed.

Reply:
libglnx has wrappers for most of the commonly-used POSIX functions now, adding `GError` support and `EINTR` hackery.